src/main.rs

```diff
@@ -62,7 +63,34 @@ async fn main() -> Result<()> {
     let url = Url::new(args.url, args.default_scheme)?;
     let host = url.host().ok_or_else(|| anyhow!("Missing hostname"))?;
     let method = args.method.unwrap_or_else(|| Method::from(&body)).into();
-    let auth = Auth::new(args.auth, args.auth_type, &host)?;
+    let mut auth = Auth::new(args.auth, args.auth_type, &host)?;
+
+    // load previous session if present
+    let previous_session = match args.session {
+        None => None,
+        Some(ref identifier) => {
+            match Session::load(identifier, &host) {
+                Err(why) => panic!("couldn't load session {}: {}", &identifier, why),
+                Ok(result) => result,
+            }
+        },
+    };
```

Comment:
Could you replace this panic with an `anyhow` error? we've been replacing all our panics with errors that are bubbled up.
```rust
use anyhow::Context;

let previous_session = match args.session {
    None => None,
    Some(ref identifier) => Session::load(identifier, &host)
        .context(format!("couldn't load session {}", identifier))?
};
```